geolocator_android/android/src/main/java/com/baseflow/geolocator/GeolocatorPlugin.java

```diff
@@ -144,6 +144,9 @@ public void onReattachedToActivityForConfigChanges(@NonNull ActivityPluginBindin
   public void onDetachedFromActivity() {
     dispose();
     deregisterListeners();
+
+    pluginBinding.getActivity().unbindService(serviceConnection);
+    pluginBinding = null;
```

Comment:
Since the `pluginBinding` can be `null` it would be good to add a null check around this block just to be on the save side.

```suggestion
    if (pluginBinding != null) {
      pluginBinding.getActivity().unbindService(serviceConnection);
      pluginBinding = null;
    }
```